web/client/plugins/Annotations.jsx

```diff
@@ -10,53 +10,41 @@ const React = require('react');
+    saveAnnotation, toggleAdd, validationError, removeAnnotationGeometry, toggleStyle, setStyle, restoreStyle,
+    highlight, cleanHighlight, showAnnotation, cancelShowAnnotation, filterAnnotations} =
     require('../actions/annotations');
-/**
-  * Annotations Plugin. Implements annotations handling on maps.
-  * Adds:
-  *  - a new vector layer, with id 'annotations', to show user created annotations on the map
-  *  - a new menu in the BurgerMenu to handle current annotations (still to be implemented)
-  *  - a custom template for Identify applied to annotations geometries that also allows editing {@link components.mapControls.annotations.AnnotationsInfoViewer}
-  *  - styling of the annotation
-  * Annotations are geometries (currently only markers are supported) with a set of properties. By default a title and
-  * a description are managed, but you can configure a different set of fields, using initialState in localConfig.json:
-  * @example
-  * {
-  *   ...
-  *   "initialState": {
-  *     "defaultState": {
-  *         "annotations": {
-  *             "fields": [{
-  *                 "name": "myattribute",
-  *                 "editable": true
-  *                 "validator": "value.indexOf('fake') === -1",
-  *                 "validateError": "annotations.error.fake"
-  *             }]
-  *         }
-  *     }
-  *   }
-  * }
-  *
-  * @class Annotations
-  * @memberof plugins
-  * @static
-  */
-const AnnotationsPlugin = connect((state) => ({
-    removing: state.annotations && state.annotations.removing,
-    editing: state.annotations && !!state.annotations.editing
-}), {
-    onCancelRemove: cancelRemoveAnnotation,
-    onConfirmRemove: confirmRemoveAnnotation
-})(require('../components/mapcontrols/annotations/Annotations'));
+
+const AnnotationsEditor = connect((state) => ({
+    config: state.annotations && state.annotations.config,
+    editing: state.annotations && state.annotations.editing,
+    drawing: state.annotations && !!state.annotations.drawing,
+    styling: state.annotations && !!state.annotations.styling,
+    errors: state.annotations.validationErrors
+}),
+{
+    onEdit: editAnnotation,
+    onCancelEdit: cancelEditAnnotation,
+    onCancel: cancelShowAnnotation,
+    onError: validationError,
+    onSave: saveAnnotation,
+    onRemove: removeAnnotation,
+    onAddGeometry: toggleAdd,
+    onStyleGeometry: toggleStyle,
+    onCancelStyle: restoreStyle,
+    onSaveStyle: toggleStyle,
+    onSetStyle: setStyle,
+    onDeleteGeometry: removeAnnotationGeometry
+})(require('../components/mapcontrols/annotations/AnnotationsEditor'));
 
 const AnnotationsInfoViewer = connect((state) => ({
-    fields: state.annotations && state.annotations.fields,
+    config: state.annotations && state.annotations.config,
     editing: state.annotations && state.annotations.editing,
     drawing: state.annotations && !!state.annotations.drawing,
     styling: state.annotations && !!state.annotations.styling,
```

Comment:
All state slices are replicated in many places. You could simply use createSelector